src/core/layertree/qgslayertreemodellegendnode.h

```diff
@@ -162,6 +162,18 @@ class CORE_EXPORT QgsSymbolV2LegendNode : public QgsLayerTreeModelLegendNode
 
     virtual void invalidateMapBasedData() override;
 
+    //! Set the icon size (icon may be smaller if croping is enabled)
```

Comment:
@wonder-sk do you mind looking at these legend related changes? ( QgsMapLayerLegend + QgsLayerTreeModelLegendNode)